ui/admin/app/controllers/scopes/scope/workers/index.js

```diff
@@ -21,6 +21,60 @@ export default class ScopesScopeWorkersIndexController extends Controller {
   // =attributes
 
   @tracked selectedWorker;
+  @tracked tags = [];
+
+  queryParams = [{ tags: { type: 'array' } }];
+
+  /**
+   * Returns all unique tags of the selected worker in base64 encoded format.
+   * @type {object}
+   */
+
+  get groupedTags() {
+    const allTags = this.model
+      .flatMap((worker) => worker.allTags)
+      .filter(Boolean);
+
+    const seen = new Set();
+    const uniqueTags = [];
+
+    for (const tag of allTags) {
+      const encodedTag = window.btoa(JSON.stringify(tag));
+
+      if (!seen.has(encodedTag)) {
+        seen.add(encodedTag);
+        uniqueTags.push(encodedTag);
+      }
+    }
+    return uniqueTags;
+  }
+
+  /**
+   * Returns the selected tags of a worker in base64 encoded format.
+   *  @type {object}
+   */
+  get encodedTags() {
+    return this.tags.map((tag) => {
+      const stringify = JSON.stringify(tag);
+      return window.btoa(stringify);
+    });
+  }
+
+  /**
+   * Returns the filters object used for displaying filter tags.
+   * @type {object}
+   */
+
+  get filters() {
+    return {
+      allFilters: {
+        tags: this.groupedTags,
```

Comment:
Rather than just sending it as is, couldn't you have transformed these into objects that match what the filters are expecting (i.e. `{ id: encodedTag, name: tag.value }`) and that way you wouldn't have to modify the filter tags component and use custom logic?

Reply:
ah yes, this is good point, I did a quick test and works as expected! I will push up the changes. I like that this won't modify the filter component. Thank you!